host-macro/src/server.rs

```diff
@@ -119,40 +125,68 @@ impl ServerBuilder {
                 /// Create a new Gatt Server instance.
                 ///
                 /// This function will add a Generic GAP Service with the given name.
-                #visibility fn new_default(stack: Stack<'reference, C>, name: &'values str) -> Self {
+                /// The maximum length which the name can be is 22 bytes (limited by the size of the advertising packet).
+                /// If a name longer than this is passed, Err() is returned.
+                #visibility fn new_default(stack: Stack<'reference, C>, name: &'values str) -> Result<Self, &'static str> {
                     let mut table: AttributeTable<'_, #mutex_type, #attribute_data_size> = AttributeTable::new();
 
-                    GapConfig::default(name).build(&mut table);
+                    static DEVICE_NAME: static_cell::StaticCell<HeaplessString<{DEVICE_NAME_MAX_LENGTH}>> = static_cell::StaticCell::new();
```

Comment:
The reason I didn't do that was because if the user attempted to create a second GapConfig then it would panic. Admittedly this implementation would panic if the user attempted to create a second instance of the type decorated with `#[gatt_server]`, but that would happen anyway due to the characteristic data stores.
If a user is only ever going to create one GapConfig then moving this code to a GapConfig method removes some duplicate code, which I'm always for.

Reply:
a6d1738830df4dd00c8493e1d542f04ec7d17d00

Reply:
Ah interesting - we were discussing this earlier and thought that creating multiple `#[gatt_servers]` in the same application wouldn't be required.

In that case I will need to revert a6d1738 as it will not allow for a second `GapConfig::Peripheral` to be made (whereas previously you could make as many `GapConfigs` as you desired but could only create one instance of a particular `#[gatt_server]` (but declaring a second `#[gatt_server]` and instantiating that would work fine).

I notice you said "putting an owned `heapless::String` directly into the `GapConfig`" rather than using a statically allocated `heapless::String` - the reason I haven't done that is that the `GapConfig` only exists for a short period of time, so the user would need to hang on to the `GapConfig` for the lifetime of the `#[gatt_server]`.

I originally tried passing ownership of the `GapConfig` to the `#[gatt_server]`, but this ends up in a chicken-and-egg situation where a reference to the `String` in the `GapConfig` is required for the attribute, but the `GapConfig` needs to be moved into the `#[gatt_server]` to complete its initialisation, thus invalidating the reference. Moving the `GapConfig` into the `#[gatt_server]` first and then adding the attributes has the same problem, as the `#[gatt_server]` is returned from the function at the end of initialisation, again causing a move and invalidating the reference.

I encountered this issue a few times during the course of this work - is there a pattern to get around it without using statics?